actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -379,7 +379,7 @@ def per_form_csrf_token(session, action_path, method)
 
       def xor_byte_strings(s1, s2)
         s2_bytes = s2.bytes
-        s1.bytes.map.with_index { |c1, i| c1 ^ s2_bytes[i] }.pack('c*')
+        s1.bytes.each_with_index { |c1, i| c1 ^ s2_bytes[i] }.pack('c*')
```

Comment:
Oh I have mistaken `Array#each_with_index`  for `Enumerator.#each_with_index` , this no need starting offset. `Enumerator.#each_with_index` also better.